File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/transaction/merge/TransactionListResponse.java
package fpt.aptech.server.dto.transaction.merge;

import fpt.aptech.server.dto.transaction.view.DailyTransactionGroup;
import lombok.Builder;
import java.math.BigDecimal;
import java.util.List;

@Builder
public record TransactionListResponse(
        BigDecimal totalIncome,           // Tổng thu
        BigDecimal totalExpense,          // Tổng chi
        BigDecimal netAmount,             // Số dư ròng (Thu - Chi)
        int transactionCount,             // Tổng số lượng giao dịch
        List<DailyTransactionGroup> dailyGroups // Danh sách nhóm theo ngày
) {
    public TransactionListResponse {
        if (totalIncome == null) totalIncome = BigDecimal.ZERO;
        if (totalExpense == null) totalExpense = BigDecimal.ZERO;
        if (netAmount == null) netAmount = BigDecimal.ZERO;
        if (dailyGroups == null) dailyGroups = List.of();
    }
}

File: APP MOBILE/smart_money/lib/modules/transaction/models/merge/transaction_list_response.dart
// modules/transaction/models/merge/transaction_list_response.dart
// Map đúng với TransactionListResponse.java của Spring Boot

import '../view/daily_transaction_group.dart';

class TransactionListResponse {
  final double totalIncome;    // Tổng thu
  final double totalExpense;   // Tổng chi
  final double netAmount;      // Số dư ròng (Thu - Chi)
  final int transactionCount;  // Tổng số lượng giao dịch
  final List<DailyTransactionGroup> dailyGroups; // Danh sách nhóm theo ngày

  const TransactionListResponse({
    required this.totalIncome,
    required this.totalExpense,
    required this.netAmount,
    required this.transactionCount,
    required this.dailyGroups,
  });

  factory TransactionListResponse.fromJson(Map<String, dynamic> json) {
    return TransactionListResponse(
      totalIncome: (json['totalIncome'] as num? ?? 0).toDouble(),
      totalExpense: (json['totalExpense'] as num? ?? 0).toDouble(),
      netAmount: (json['netAmount'] as num? ?? 0).toDouble(),
      transactionCount: json['transactionCount'] as int? ?? 0,
      dailyGroups: json['dailyGroups'] != null
          ? (json['dailyGroups'] as List<dynamic>)
          .map((e) => DailyTransactionGroup.fromJson(e as Map<String, dynamic>))
          .toList()
          : [],
    );
  }
}